MovieQuiz/Presentation/MovieQuizPresenter.swift

```diff
@@ -0,0 +1,158 @@
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [weak self] in
+            guard let self = self else { return }
+            self.proceedToNextQuestionOrResults()
+        }
+    }
+    
+    private func proceedToNextQuestionOrResults() {
+        if self.isLastQuestion() {
+          let viewModel = QuizResultsViewModel(
+            title: "Этот раунд окончен!",
+            text: "",
+            buttonText: "Сыграть ещё раз")
+          
+            viewController?.show(quiz: viewModel)
+          
+      } else {
+          self.switchToNextQuestion()
+          questionFactory?.requestNextQuestion()
+      }
+    }
+    
+    // MARK: - QuestionFactoryDelegate
+    
+    func didRecieveNextQuestion(question: QuizQuestion?) {
+        guard let question = question else { return }
+        
+        currentQuestion = question
+        let viewModel = convert(model: question)
+        
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            self.viewController?.show(quiz: viewModel)
+
+        }
+        viewController?.hideLoadingIndicator()
+    }
+    
+    func didLoadDataFromServer() {
+        viewController?.showLoadingIndicator()
+        questionFactory?.requestNextQuestion()
+    }
+    
+    func didFailToLoadData(with error: Error) {
+        viewController?.showNetworkError(message: error.localizedDescription)
+    }
+    
+    // Для ошибок: при неуспешной загрузке измененного изображения по ссылке и при наличии errorMessage у MostPopularMovies
+    func didFailToLoad(message: String) {
+        viewController?.showNetworkError(message: message)
+    }
+    
+    // Для отображения индикатора, пока загружается измененное изображение по ссылке
+    func resizedImageLoading() {
+        viewController?.showLoadingIndicator()
+    }
+    
+    // MARK: - Functions
+    
+    func convert(model: QuizQuestion) -> QuizStepViewModel {
+        return QuizStepViewModel(
+            image: UIImage(data: model.image) ?? UIImage(),
+            question: model.text,
+            questionNumber: "\(currentQuestionIndex + 1)/\(questionsAmount)")
+    }
+    
+    func restartGame() {
+        correctAnswers = 0
+        resetQuestionIndex()
+        questionFactory?.requestNextQuestion()
+    }
+    
+    func noButtonClicked() {
+        didAnswer(isYes: false)
+    }
+    
+    func yesButtonClicked() {
+        didAnswer(isYes: true)
+    }
+    
+    func makeResultsMessage() -> String {
+        statisticService.store(correct: correctAnswers, total: questionsAmount)
+        
+        let text = """
+  Ваш результат: \(correctAnswers) из \(questionsAmount)
+  Количество сыгранных квизов: \(statisticService.gamesCount)
+  Рекорд: \(statisticService.bestGame.correct)/\(statisticService.bestGame.total) (\(statisticService.bestGame.date.dateTimeString))
+  Средняя точность: \(String(format: "%.2f", statisticService.totalAccuracy))%
+  """
+        return text
```

Comment:
Так тоже будет работать, при этом выглядеть будет аккуратнее
```suggestion
        let text = """
        Ваш результат: \(correctAnswers) из \(questionsAmount)
        Количество сыгранных квизов: \(statisticService.gamesCount)
        Рекорд: \(statisticService.bestGame.correct)/\(statisticService.bestGame.total) (\(statisticService.bestGame.date.dateTimeString))
        Средняя точность: \(String(format: "%.2f", statisticService.totalAccuracy))%
        """
        return text
```